rust/libnewsboat-ffi/src/configpaths.rs

```diff
@@ -0,0 +1,140 @@
+use abort_on_panic;
+use libc::{c_char, c_void};
+use libnewsboat::cliargsparser::CliArgsParser;
+use libnewsboat::configpaths::ConfigPaths;
+use std::ffi::{CStr, CString};
+use std::mem;
+use std::panic::{RefUnwindSafe, UnwindSafe};
+use std::ptr;
+
+#[no_mangle]
+pub extern "C" fn create_rs_configpaths() -> *mut c_void {
+    abort_on_panic(|| Box::into_raw(Box::new(ConfigPaths::new())) as *mut c_void)
+}
+
+#[no_mangle]
+pub extern "C" fn destroy_rs_configpaths(object: *mut c_void) {
+    abort_on_panic(|| {
+        if object.is_null() {
+            return;
+        };
+        unsafe {
+            Box::from_raw(object);
+        }
+    })
+}
+
+fn with_configpaths<F, T>(object: *mut c_void, action: F, default: T) -> T
+where
+    F: RefUnwindSafe + Fn(&mut Box<ConfigPaths>) -> T,
```

Comment:
This should be:
```
     F: RefUnwindSafe + Fn(&mut ConfigPaths) -> T,
```
This is so the function can contain the content of the Box, but not the Box itself. In case of `box_a: &mut Box<ConfigPaths>` a function could assign a new Box to `box_a` which would cause a free of the old box, while C is still using it as a pointer.